assets/scripts/cdocs-build.js

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env node
+
+/**
+ * This script runs as part of the build, before Hugo has processed any files.
+ * It invokes the Markdoc integration to compile all .mdoc files
+ * found in the content directory, and updates the .gitignore file
+ * in that directory to ignore the compiled files.
+ */
+console.time('Markdoc compilation execution time');
+
+const fs = require('fs');
+const path = require('path');
+
+const { CdocsHugoIntegration } = require('cdocs-hugo-integration');
+
+const env = process.env.CI_ENVIRONMENT_NAME || 'development';
+const baseSiteDir = path.resolve(__dirname, '../..');
+
+const ASSETS_PARTIAL_PATH = path.resolve(__dirname, `${baseSiteDir}/layouts/partials/markdoc-assets.html`);
+const CONTENT_DIR = path.resolve(__dirname, `${baseSiteDir}/content`);
+
+// Initialize the Markdoc integration
+const markdocIntegration = new CdocsHugoIntegration({
+    baseSiteDir,
+    env
+});
+
+// Build the assets partial, and write it to the target file path
+const assetsPartialContents = markdocIntegration.buildAssetsPartial();
+fs.writeFileSync(ASSETS_PARTIAL_PATH, assetsPartialContents);
+
+// Compile all .mdoc files found in the content directory
+const { compiledFilePaths, hasErrors } = markdocIntegration.compileMdocFiles();
+
+console.log(`Markdoc compilation completed. Compiled ${compiledFilePaths.length} files.`);
+
+// Build a .gitignore file for the compiled files,
+// to be written to the content directory
+let gitignoreContents = `# GENERATED FILE: DO NOT EDIT
+# Ignore .md files compiled from Markdoc\n`;
+
+compiledFilePaths.forEach((file) => {
```

Comment:
If we're generating a fresh list each time and writing the file at runtime, could we run into an issue where we remove an mdoc file but forget to remove the markdown? In that case, the file might accidentally show up in a commit and be published accidentally.

Perhaps we should check the list of files against the existing lines in .gitignore and add new entries instead of generating a fresh list each time?